s3auth-relay/src/main/java/com/s3auth/relay/FtpFacade.java

```diff
@@ -110,75 +67,25 @@
      */
     FtpFacade(@NotNull final Hosts hosts, final int port)
         throws IOException {
-        this.server = new ServerSocket(port);
-        final FtpThread thread = new FtpThread(this.sockets, hosts);
-        final Runnable runnable = new VerboseRunnable(
-            new FtpFacade.FTPThreadRunnable(thread), true, true
-        );
-        for (int idx = 0; idx < FtpFacade.THREADS; ++idx) {
-            this.backend.scheduleWithFixedDelay(
-                runnable,
-                0L, 1L, TimeUnit.NANOSECONDS
-            );
-        }
+        this.setFrontend(this.createThreadPool(2, "FTP-front"));
```

Comment:
@simonjenga instead of `setFrontend`, let's add a protected constructor to the parent class that takes as parameters `2` and "FTP-front".